### PR TITLE
Update build.gradle

### PR DESCRIPTION
**Pull request by xiaoyongzhu**
## Description

Resolves #XXX

## Does this PR introduce any user-facing changes?

- [ ] No. You can skip the rest of this section.
- [ ] Yes. Make sure to clarify your proposed changes.

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -57,9 +57,8 @@ jar {
   }
   exclude 'META-INF/*.RSA', 'META-INF/*.SF','META-INF/*.DSA'
 
-  // Explicitly exclude com/linkedin/data and org/apache/hadoop files from the final jar. They can cause issues in other downstream applications.
+  // Explicitly exclude com/linkedin/data files from the final jar. They can cause issues in other downstream applications.
   exclude 'com/linkedin/data/**'
-  exclude 'org/apache/hadoop/**'
```
Comment (rakeshkashyap123) [inline build.gradle:62]: Is this intended? Do we need the hadoop files in the fat jar?

Comment (xiaoyongzhu) [reply build.gradle:62]: Good point, I'm not sure. It's a test run to validate @jainr 's thought.

**Comment by xiaoyongzhu** [comment]
https://docs.gradle.org/current/userguide/java_plugin.html#sec:java_plugin_and_dependency_management This might be a useful reference

**Comment by xiaoyongzhu** [comment]
I'm not a gradle expert but per the docs above, maybe we can try something:
`implementation`, `runtimeClasspath`, `archives`, `default`

implementation
Implementation only dependencies.

compileOnly
Compile time only dependencies, not used at runtime.

compileClasspath extends compileOnly, implementation
Compile classpath, used when compiling source. Used by task compileJava.

annotationProcessor
Annotation processors used during compilation.

runtimeOnly
Runtime only dependencies.

runtimeClasspath extends runtimeOnly, implementation
Runtime classpath contains elements of the implementation, as well as runtime only elements.

testImplementation extends implementation
Implementation only dependencies for tests.

testCompileOnly
Additional dependencies only for compiling tests, not used at runtime.

testCompileClasspath extends testCompileOnly, testImplementation
Test compile classpath, used when compiling test sources. Used by task compileTestJava.

testRuntimeOnly extends runtimeOnly
Runtime only dependencies for running tests.

testRuntimeClasspath extends testRuntimeOnly, testImplementation
Runtime classpath for running tests. Used by task test.

archives
Artifacts (e.g. jars) produced by this project. Used by Gradle to determine "default" tasks to execute when building.

default extends runtimeElements
The default configuration used by a project dependency on this project. Contains the artifacts and dependencies required by this project at runtime.

**Comment by jainr** [comment]
Thanks for the suggestion, if you look at my latest PR #947 you will notice I am past the class not found failure by following some of the above concepts. It's complaining now about a missing setting.

**Pull request closed without merging**